lib/Map/LeafletVisualizer.js

```diff
@@ -185,6 +186,41 @@ LeafletGeomVisualizer.prototype.update = function(time) {
     return true;
 };
 
+/**
+ * Computes the rectangular bounds which encloses the collection of
+ * entities to be visualized.
+ *
+ * @returns {LatLngBounds} The computed bounds.
+ */
+LeafletGeomVisualizer.prototype.getLatLngBounds = function() {
+    var entities = this._entitiesToVisualize.values;
+    var bounds = [];
+
+    for (var i = 0, len = entities.length; i < len; i++) {
+        var entity = entities[i];
+        var latLngs;
+
+        if (defined(entity._point) || defined(entity._billboard) || defined(entity._label)) {
```

Comment:
This works, and is nicely written, but I think it would be better to let Leaflet do more of the work. Every Leaflet layer has a `getBounds` method. So `this._entityHash` is a hash of entity IDs to "details" objects. The details object has properties like `point`, `polygon`, etc. with details of that type of graphical primitive associated with that entity. Then each graphical primitive object as a `layer` property which is a Leaflet layer. Call `getBounds` on each layer, `extend` them into a single `LatLngBnds`, and you have your answer! 

You can write code that does this without even knowing the different types of graphical elements. E.g. something like:

```
Object.keys(this._entityHash).forEach(entityId => {
    const entityDetails = this._entityHash[entityId];
    Object.keys(entityDetails).forEach(primitiveId => {
        const primitive = entityDetails[primitiveId];
        if (primitive.layer !== undefined && primitive.layer.getBounds !== undefined) {
            const bounds = primitive.layer.getBounds();
            // combine bounds with overall bounds with extend
        }
    });
});